gcloud-java-dns/src/test/java/com/google/gcloud/testing/LocalDnsHelperTest.java

```diff
@@ -0,0 +1,955 @@
+import java.util.List;
+import java.util.Map;
+
+public class LocalDnsHelperTest {
+
+  private static final String RRSET_TYPE = "A";
+  private static final ResourceRecordSet RRSET1 = new ResourceRecordSet();
+  private static final ResourceRecordSet RRSET2 = new ResourceRecordSet();
+  private static final ResourceRecordSet RRSET_KEEP = new ResourceRecordSet();
+  private static final String PROJECT_ID1 = "2135436541254";
+  private static final String PROJECT_ID2 = "882248761325";
+  private static final String ZONE_NAME1 = "my little zone";
+  private static final String ZONE_NAME2 = "another zone name";
+  private static final ManagedZone ZONE1 = new ManagedZone();
+  private static final ManagedZone ZONE2 = new ManagedZone();
+  private static final String DNS_NAME = "www.example.com.";
+  private static final Change CHANGE1 = new Change();
+  private static final Change CHANGE2 = new Change();
+  private static final Change CHANGE_KEEP = new Change();
+  private Map<String, Object> optionsMap;
+  private LocalDnsHelper localDns;
+  private ManagedZone minimalZone = new ManagedZone(); // to be adjusted as needed
+
+  @BeforeClass
+  public static void before() {
+    RRSET1.setName(DNS_NAME);
+    RRSET1.setType(RRSET_TYPE);
+    RRSET1.setRrdatas(ImmutableList.of("1.1.1.1"));
+    ZONE1.setName(ZONE_NAME1);
+    ZONE1.setDescription("");
+    ZONE1.setDnsName(DNS_NAME);
+    ZONE2.setName(ZONE_NAME2);
+    ZONE2.setDescription("");
+    ZONE2.setDnsName(DNS_NAME);
+    RRSET2.setName(DNS_NAME);
+    RRSET2.setType(RRSET_TYPE);
+    RRSET_KEEP.setName(DNS_NAME);
+    RRSET_KEEP.setType("MX");
+    RRSET_KEEP.setRrdatas(ImmutableList.of("255.255.255.254"));
+    RRSET2.setRrdatas(ImmutableList.of("123.132.153.156"));
+    CHANGE1.setAdditions(ImmutableList.of(RRSET1, RRSET2));
+    CHANGE2.setDeletions(ImmutableList.of(RRSET2));
+    CHANGE_KEEP.setAdditions(ImmutableList.of(RRSET_KEEP));
+  }
+
+  @Before
+  public void setUp() {
+    localDns = LocalDnsHelper.create(0L); // synchronous
+    optionsMap = new HashMap<>();
+    minimalZone = copyZone(ZONE1);
+  }
+
+  @After
+  public void after() {
+    localDns = null;
+  }
+
+  @Test
+  public void testMatchesCriteria() {
+    assertTrue(LocalDnsHelper.matchesCriteria(RRSET1, RRSET1.getName(), RRSET1.getType()));
+    assertFalse(LocalDnsHelper.matchesCriteria(RRSET1, RRSET1.getName(), "anothertype"));
+    assertTrue(LocalDnsHelper.matchesCriteria(RRSET1, null, RRSET1.getType()));
+    assertTrue(LocalDnsHelper.matchesCriteria(RRSET1, RRSET1.getName(), null));
+    assertFalse(LocalDnsHelper.matchesCriteria(RRSET1, "anothername", RRSET1.getType()));
+  }
+
+  @Test
+  public void testGetUniqueId() {
+    assertNotNull(LocalDnsHelper.getUniqueId(Lists.<LocalDnsHelper.RrsetWrapper>newLinkedList()));
+  }
+
+  @Test
+  public void testFindProject() {
+    assertEquals(0, localDns.projects().size());
+    LocalDnsHelper.ProjectContainer project = localDns.findProject(PROJECT_ID1);
```

Comment:
Ok, I agree then. However, since this mainly concerns future maintenance, I am going to set up an issue for this and clean this up separately as soon as the helper itself is LGTM'd. The tests might still come in handy to me if there is any bigger refactoring to be done now. 

Opened #671.